scipy/signal/wigner.py

```diff
@@ -0,0 +1,99 @@
+"""Tools based on the Wigner quasiprobability distributions.
+"""
+
+import numpy as np
+from scipy.fft import fft
+
+
+__all__ = ['wvd']
+
+
+def wigner_ville(x, fs=1.0, resolution=1, window_size=None):
+    """
+    Wigner-Ville distribution.
+
+    The Wigner–Ville distribution is a method of signal analysis that yields
+    (under ideal conditions) the signal energy distributed over time and
+    frequency. It provides the highest temporal vs frequency resolution
+    possible within the limitations of the uncertainty principle.
+
+    Parameters
+    ----------
+    x : array_like
+        Time series of measurement values.
+    fs : float, optional
+        Sampling frequency of the `x` time series. Defaults to 1.0.
+    resolution : int, optional
+        Resolution, number of samples between windows. Defaults to 1.
+    window_size : int, optional
+        Window size, length of frequency axis. Defaults to ``len(x) // 2``.
+
+    Returns
+    -------
+    t : ndarray
+        Array of time samples.
+    f : ndarray
+        Array of frequency samples.
+    wv : ndarray
+        Wigner-Ville distribution of `x`.
+
+    References
+    ----------
+    .. [1] S. Mallat, "A Wavelet Tour of Signal Processing (3rd Edition)",
+        Academic Press, 2009.
+
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy import signal
+
+    Generate a test signal with two superimposed sine wave at 1 Hz and 5 Hz.
+
+    >>> f_s = 100
+    >>> t = np.arange(0, 1024) / 100
+    >>> x = np.sin(2*np.pi*1*t) + np.sin(2*np.pi*5*t)
+
+    Compute the Wigner-Ville distribution. Note that the analystic signal is
+    supplied to the function.
+
+    >>> t, f, wv = signal.wigner_ville(signal.hilbert(x), f_s)
+
+    Plot the Wigner-Ville distribution with expected cross-terms clearly
+    visible at 3 Hz.
+
+    >>> plt.figure()
+    >>> plt.pcolormesh(t, f, wv, shading='nearest')
+    >>> plt.xlabel('Time $t$ / s')
+    >>> plt.ylabel('Frequency $f$ / Hz')
+    >>> plt.ylim([0, 6])
+    >>> plt.show()
+    """
+    if not window_size:
+        window_size = len(x) // 2
```

Comment:
```suggestion
    x = np.asarray(x)
    if not window_size:
        window_size = len(x) // 2
```
With the large variety of different array-likes that's out there, there is no guarantee that `x` is a `Sized` object, 
even if it does have the correct dimensions. Explicitly calling `np.asarray` beforehand should resolve these 
(somewhat niche) cases.

Reply:
``` diff
- if not window_size:
+ if window_size is not None:
```
Unless someone is willing to argue that `None` and `0` should be treated as equivalent, 
I'd also suggest to perform an explicit `is None` check rather than truth-testing.

Reply:
> With the large variety of different array-likes that's out there, there is no guarantee that `x` is a `Sized` object,
> even if it does have the correct dimensions. Explicitly calling `np.asarray` beforehand should resolve these
> (somewhat niche) cases.

You never know what people come up with. Done.


> ```diff
> - if not window_size:
> + if window_size is not None:
> ```
No need to argue, but I guess it's supposed to be `if window_size is None`.